app/assets/javascripts/service-worker.js.erb

```diff
@@ -126,6 +125,22 @@ self.addEventListener('pushsubscriptionchange', function(event) {
   );
 });
 
+self.addEventListener('message', function(event) {
+  if (!event.data.chatSound) {
+    return;
+  }
+
+  event.waitUntil(
+    self.clients.matchAll().then(function(clients) {
+      clients.forEach(function(client) {
+        client.postMessage({
+          canPlaySound: client.visibilityState === "visible"
```

Comment:
We iterate over all open clients and post a response back to the browser. It's only a true value for the active browser tab.

Reply:
Out of curiosity, does the sound play when the browser is minimized / in the background?

Reply:
If it is in the background then it shouldn't play because we will return a false value, however minimized is a good question. I think it will likely still be "hidden" in that case and return false.

We also have the option of checking `client.focused` which is a boolean value too, but from experimenting it always seemed to match the visibilityState.